sphinxdocs/src/sphinx_stardoc/stardoc.py

```diff
@@ -608,25 +620,52 @@ def make_xref(name):
             sig_node += attr_annotation_node
 
         if params_text:
-            signature = inspect.signature_from_str(params_text)
-
-            paramlist_node = addnodes.desc_parameterlist()
-            for param in signature.parameters.values():
-                node = addnodes.desc_parameter()
-                node += addnodes.desc_sig_name(rawsource="", text=param.name)
-                if param.default is not param.empty:
-                    node += addnodes.desc_sig_operator("", "=")
-                    node += docutils_nodes.inline(
-                        "",
-                        param.default,
-                        classes=["default_value"],
-                        support_smartquotes=False,
-                    )
-                paramlist_node += node
-            sig_node += paramlist_node
-
-            if signature.return_annotation is not signature.empty:
-                sig_node += addnodes.desc_returns("", signature.return_annotation)
+            try:
+                signature = inspect.signature_from_str(params_text)
+            except SyntaxError:
+                # Stardoc doesn't provide accurate info, so the reconstructed
+                # signature might not be valid syntax. Rather than fail, just
+                # provide a plain-text description of the approximate signature.
```

Comment:
Is it something that should be reported upstream?

Reply:
Yes. I added a comment here to reference https://github.com/bazelbuild/stardoc/issues/225